src/Laravel/ProvisionServiceProvider.php

```diff
@@ -95,8 +95,10 @@ protected function bootRegistry(): void
         // Attempt to set the Registry instance from cache
         if ($cachedRegistry = $cache->get(self::REGISTRY_CACHE_KEY)) {
             try {
-                $registry = unserialize($cachedRegistry);
-            } catch (Throwable $e) {
+                $registry = unserialize($cachedRegistry, ['allowed_classes' => [Registry::class]]);
```

Comment:
Am I right in thinking this is a more complex object graph of categories, each with their own classes and types? I'd have thought we'd need to add more types to this allowlist?

Reply:
Hm, not really sure, I think it just needs the class that is serialized (regardless of what's included). Not entirely sure, might need to play around a bit more
PHPStorm flagged it like it should fixed, hence added.

Reply:
https://3v4l.org/kieEO

It seems to unserialize but it will create a `__PHP_Incomplete_Class`. If you add a type-hint on `B` in that linked example it'll error out.

Reply:
So, if we want to use that, we will need to add all the potential classes? hm, that's not really helpful.

I think I need to set it to `true` then to serialize all.
Thanks for flagging that!

Reply:
Yeh since its internally cached and unserialized im not concerned about a security risk of allowing all classes